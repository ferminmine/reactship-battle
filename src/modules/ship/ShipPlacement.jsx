import React from 'react';
import Board from '../board/Board';

const placeShip = ({ column, row }) => console.log(`${column} ${row}`);

const ShipPlacement = () => <Board actionToPerform={placeShip} />;

export default ShipPlacement;