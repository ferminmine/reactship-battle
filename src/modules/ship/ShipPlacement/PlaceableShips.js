const AVAILABLE_SHIPS = {
  submarine: {
    name: 'Submarine',
    size: 2,
    image: '/imgs/submarine.png'
  },
  battleShip: {
    name: 'Battleship',
    size: 3,
    image: '/imgs/battleship.png'
  },
  cruiser: {
    name: 'Cruiser',
    size: 4,
    image: '/imgs/cruiser.png'
  }
};

const placeableShips = [
  AVAILABLE_SHIPS.cruiser,
  AVAILABLE_SHIPS.battleShip,
  AVAILABLE_SHIPS.battleShip,
  AVAILABLE_SHIPS.battleShip,
  AVAILABLE_SHIPS.submarine
];

export default placeableShips;
