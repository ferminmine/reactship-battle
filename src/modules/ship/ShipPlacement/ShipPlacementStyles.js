export default {
  buttonContainer: {
    display: 'flex',
    alignItems: 'center'
  },
  rotateButton: {
    background: 'none',
    border: '3px solid #2f9ad4',
    color: '#2f9ad4',
    fontWeight: '800',
    padding: '4px',
    borderRadius: '6px',
    fontSize: '1.1em',
    cursor: 'pointer',
    marginRight: '22%',
    textAlign: 'center'
  },
  boardOutsideContainer: {
    width: '25%'
  },
  '@media (max-width: 650px)': {
    boardOutsideContainer: {
      width: '90%',
      marginTop: '2em'
    }
  }
};
