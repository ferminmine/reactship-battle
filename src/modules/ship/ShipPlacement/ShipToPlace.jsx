import React from 'react';
import withStyles from 'react-jss';
import { compose } from 'redux';
import { connect } from 'react-redux';
import { nextShipToPlaceSelector } from '../ShipSelectors';

const styles = {
    shipToPlaceContainer: {
      marginBottom: '2em',
      textAlign: 'center'
    },
    nextShip: {
      color: '#2f9ad4'
    },
    shipName: {
      color: '#7b7979',
      fontWeight: 300,
      marginTop: 0
    },
    shipImage: {
      height: '5em'
    }
}

class ShipToPlace extends React.Component {
    render = () => {
        const nextShip = this.props.nextShipToPlace && this.props.shipsToPlace[0];
        const { classes } = this.props;
        return (
          <div className={classes.shipToPlaceContainer}>
            <h1 className={classes.nextShip}> 
              { nextShip !== undefined ? 'Next Ship to Place' : 'No more ships available' } 
            </h1>
            { 
              nextShip !== undefined && ( 
              <React.Fragment>
                <img className={classes.shipImage} alt='' src={nextShip && nextShip.image} />
                <h2 className={classes.shipName}> {nextShip && nextShip.name} </h2>
              </React.Fragment>
              )}
          </div>
        )
    }
}

const mapStateToProps = state => ({
  shipsToPlace: state.ships.shipsToPlace,
  nextShipToPlace: nextShipToPlaceSelector(state)
})

export default compose(
  withStyles(styles),
  connect(mapStateToProps)
)(ShipToPlace);