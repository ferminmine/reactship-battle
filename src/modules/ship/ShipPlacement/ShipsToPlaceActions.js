export const SHIPS_TO_PLACE_ACTION_TYPES = {
  REMOVE_SHIP_TO_PLACE: 'REMOVE_SHIP_TO_PLACE'
};

export const removeShipToPlace = () => ({
  type: SHIPS_TO_PLACE_ACTION_TYPES.REMOVE_SHIP_TO_PLACE
});
