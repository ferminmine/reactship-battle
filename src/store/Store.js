import { createStore, combineReducers, compose, applyMiddleware } from 'redux';
import { createBrowserHistory } from 'history';
import { connectRouter, routerMiddleware } from 'connected-react-router';
import playerReducer from '../modules/player/PlayerReducer';
import shipReducer from '../modules/ship/ShipReducer';
import gameReducer from '../modules/game/GameReducer';
import shootsReducer from '../modules/shoots/ShootsReducer';

export const history = createBrowserHistory();

const allReducers = combineReducers({
  player: playerReducer,
  game: gameReducer,
  ships: shipReducer,
  shoots: shootsReducer,
  router: connectRouter(history)
});

export const ROOT_REDUCER_ACTION_TYPES = {
  RESET_APP: 'RESET_APP'
};
export const resetApp = () => ({ type: ROOT_REDUCER_ACTION_TYPES.RESET_APP });

const rootReducer = (state, action) => {
  if (action.type === 'RESET_APP') {
    state = undefined;
  }

  return allReducers(state, action);
};

const composeEnhancer = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

export default createStore(
  rootReducer,
  composeEnhancer(applyMiddleware(routerMiddleware(history)))
);
