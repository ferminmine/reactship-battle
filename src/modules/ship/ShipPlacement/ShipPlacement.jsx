import React from 'react';
import Board from '../../board/Board';
import { connect } from 'react-redux';
import { compose, bindActionCreators } from 'redux';
import { setOrientation, addShip } from '../ShipsActions';
import { removeShipToPlace } from './ShipsToPlaceActions';
import withStyles from 'react-jss';
import PropTypes from 'prop-types';
import { nextShipToPlaceSelector } from '../ShipSelectors';
import positionValidator from '../../../utils/PositionValidator';
import shipCollapseValidator from '../../../utils/ShipCollapseValidator';

const styles = {
    buttonContainer: {
        display: 'flex',
        alignItems: 'center'
    },
    rotateButton: {
        background: 'none',
        border: '3px solid #2f9ad4',
        color: '#2f9ad4',
        fontWeight: '800',
        padding: '4px',
        borderRadius: '6px',
        fontSize: '1.1em',
        cursor: 'pointer',
        marginRight: '22%',
        textAlign: 'center'
    }
}

class ShipPlacement extends React.Component {
  static propTypes = {
    orientation: PropTypes.string,
    nextShipToPlace: PropTypes.object,
    setOrientation: PropTypes.func.isRequired,
    addShip: PropTypes.func.isRequired,
    removeShipToPlace: PropTypes.func.isRequired,
    classes: PropTypes.object
  };

  placeShip = ({ column, row }) => {
    const { nextShipToPlace, orientation } = this.props;
    if (nextShipToPlace && positionValidator(orientation, nextShipToPlace.size, column, row)) {
      let shipToCreate = {
        ...nextShipToPlace,
        position: { column, row },
        orientation,
        currentLife: 4
      };
      if (!shipCollapseValidator(shipToCreate, this.props.playerShips)) {
        this.props.addShip(shipToCreate, 'HUMAN');
        this.props.removeShipToPlace();
      } else {
        alert('Collapsing!');
      }
    }
  };

  changeOrientation = event => {
    let newOrientation = this.props.orientation === 'horizontal' ? 'vertical' : 'horizontal';
    this.props.setOrientation(newOrientation);
  };

  componentDidMount = () => {
    document.addEventListener('keydown', this.changeOrientation, false);
  };

  render = () => {
    const { classes } = this.props;  
    return (
      <React.Fragment>
          <div className={classes.buttonContainer}>
            <div className={classes.rotateButton} onClick={this.changeOrientation}> ROTATE SHIP </div>
          </div>
          <Board
            type={'shipPlacer'}
            actionToPerform={this.placeShip}
            nextShipToPlace={this.props.nextShipToPlace}
            ships={this.props.playerShips}
          />
      </React.Fragment>)
  };
}

const mapStateToProps = state => ({
  orientation: state.ships.orientationToPlace,
  nextShipToPlace: nextShipToPlaceSelector(state),
  playerShips: state.ships.playerShips
});

const mapDispatchToProps = dispatch =>
  bindActionCreators(
    {
      setOrientation,
      removeShipToPlace,
      addShip
    },
    dispatch
  );

export default compose(
    withStyles(styles),
    connect(mapStateToProps, mapDispatchToProps)
)(ShipPlacement);
