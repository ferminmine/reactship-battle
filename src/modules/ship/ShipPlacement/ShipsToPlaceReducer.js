import placeableShips from './PlaceableShips';
import { SHIPS_TO_PLACE_ACTION_TYPES } from './ShipsToPlaceActions';

const defaultState = [...placeableShips];

const shipsToPlaceReducer = (state = defaultState, action) => {
  switch (action.type) {
    case SHIPS_TO_PLACE_ACTION_TYPES.REMOVE_SHIP_TO_PLACE:
      state.shift();
      return [...state];
    default:
      return state;
  }
};

export default shipsToPlaceReducer;
