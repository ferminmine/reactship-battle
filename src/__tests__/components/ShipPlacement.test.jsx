import React from 'react';
import { mount } from 'enzyme';
import ShipPlacement from '../../modules/ship/ShipPlacement/ShipPlacement';
import { Provider } from 'react-redux';
import store from '../../store/Store';
import renderer from 'react-test-renderer';

describe('ShipPlacement Component Test', () => {
  it('renders without crashing', () => {
    mount(
      <Provider store={store}>
        {' '}
        <ShipPlacement />{' '}
      </Provider>
    );
  });

  it('matches the snapshot', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <ShipPlacement />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
