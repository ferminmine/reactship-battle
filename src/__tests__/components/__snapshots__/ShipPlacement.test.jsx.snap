// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShipPlacement Component Test matches the snapshot 1`] = `
Array [
  <div
    className="Connect(ShipPlacement)-buttonContainer-0-1-1"
  >
    <div
      className="Connect(ShipPlacement)-rotateButton-0-1-2"
      onClick={[Function]}
    >
       ROTATE SHIP 
    </div>
  </div>,
  <div
    className="Connect(ShipPlacement)-boardOutsideContainer-0-1-3"
  >
    <div
      className="Connect(Board)-boardContainer-0-1-4"
    >
      <div
        className="Connect(Board)-land-0-1-5 undefined undefined undefined undefined undefined undefined"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      />
      <div
        className="Connect(Board)-land-0-1-5 undefined undefined undefined undefined undefined undefined"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      />
      <div
        className="Connect(Board)-land-0-1-5 undefined undefined undefined undefined undefined undefined"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      />
      <div
        className="Connect(Board)-land-0-1-5 undefined undefined undefined undefined undefined undefined"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      />
      <div
        className="Connect(Board)-land-0-1-5 undefined undefined undefined undefined undefined undefined"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      />
      <div
        className="Connect(Board)-land-0-1-5 undefined undefined undefined undefined undefined undefined"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      />
      <div
        className="Connect(Board)-land-0-1-5 undefined undefined undefined undefined undefined undefined"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      />
      <div
        className="Connect(Board)-land-0-1-5 undefined undefined undefined undefined undefined undefined"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      />
      <div
        className="Connect(Board)-land-0-1-5 undefined undefined undefined undefined undefined undefined"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      />
      <div
        className="Connect(Board)-land-0-1-5 undefined undefined undefined undefined undefined undefined"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      />
      <div
        className="Connect(Board)-land-0-1-5 undefined undefined undefined undefined undefined undefined"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      />
      <div
        className="Connect(Board)-land-0-1-5 undefined undefined undefined undefined undefined undefined"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      />
      <div
        className="Connect(Board)-land-0-1-5 undefined undefined undefined undefined undefined undefined"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      />
      <div
        className="Connect(Board)-land-0-1-5 undefined undefined undefined undefined undefined undefined"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      />
      <div
        className="Connect(Board)-land-0-1-5 undefined undefined undefined undefined undefined undefined"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      />
      <div
        className="Connect(Board)-land-0-1-5 undefined undefined undefined undefined undefined undefined"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      />
      <div
        className="Connect(Board)-land-0-1-5 undefined undefined undefined undefined undefined undefined"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      />
      <div
        className="Connect(Board)-land-0-1-5 undefined undefined undefined undefined undefined undefined"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      />
      <div
        className="Connect(Board)-land-0-1-5 undefined undefined undefined undefined undefined undefined"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      />
      <div
        className="Connect(Board)-land-0-1-5 undefined undefined undefined undefined undefined undefined"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      />
      <div
        className="Connect(Board)-land-0-1-5 undefined undefined undefined undefined undefined undefined"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      />
      <div
        className="Connect(Board)-land-0-1-5 undefined undefined undefined undefined undefined undefined"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      />
      <div
        className="Connect(Board)-land-0-1-5 undefined undefined undefined undefined undefined undefined"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      />
      <div
        className="Connect(Board)-land-0-1-5 undefined undefined undefined undefined undefined undefined"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      />
      <div
        className="Connect(Board)-land-0-1-5 undefined undefined undefined undefined undefined undefined"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      />
      <div
        className="Connect(Board)-land-0-1-5 undefined undefined undefined undefined undefined undefined"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      />
      <div
        className="Connect(Board)-land-0-1-5 undefined undefined undefined undefined undefined undefined"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      />
      <div
        className="Connect(Board)-land-0-1-5 undefined undefined undefined undefined undefined undefined"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      />
      <div
        className="Connect(Board)-land-0-1-5 undefined undefined undefined undefined undefined undefined"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      />
      <div
        className="Connect(Board)-land-0-1-5 undefined undefined undefined undefined undefined undefined"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      />
      <div
        className="Connect(Board)-land-0-1-5 undefined undefined undefined undefined undefined undefined"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      />
      <div
        className="Connect(Board)-land-0-1-5 undefined undefined undefined undefined undefined undefined"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      />
      <div
        className="Connect(Board)-land-0-1-5 undefined undefined undefined undefined undefined undefined"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      />
      <div
        className="Connect(Board)-land-0-1-5 undefined undefined undefined undefined undefined undefined"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      />
      <div
        className="Connect(Board)-land-0-1-5 undefined undefined undefined undefined undefined undefined"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      />
      <div
        className="Connect(Board)-land-0-1-5 undefined undefined undefined undefined undefined undefined"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      />
      <div
        className="Connect(Board)-land-0-1-5 undefined undefined undefined undefined undefined undefined"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      />
      <div
        className="Connect(Board)-land-0-1-5 undefined undefined undefined undefined undefined undefined"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      />
      <div
        className="Connect(Board)-land-0-1-5 undefined undefined undefined undefined undefined undefined"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      />
      <div
        className="Connect(Board)-land-0-1-5 undefined undefined undefined undefined undefined undefined"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      />
      <div
        className="Connect(Board)-land-0-1-5 undefined undefined undefined undefined undefined undefined"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      />
      <div
        className="Connect(Board)-land-0-1-5 undefined undefined undefined undefined undefined undefined"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      />
      <div
        className="Connect(Board)-land-0-1-5 undefined undefined undefined undefined undefined undefined"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      />
      <div
        className="Connect(Board)-land-0-1-5 undefined undefined undefined undefined undefined undefined"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      />
      <div
        className="Connect(Board)-land-0-1-5 undefined undefined undefined undefined undefined undefined"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      />
      <div
        className="Connect(Board)-land-0-1-5 undefined undefined undefined undefined undefined undefined"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      />
      <div
        className="Connect(Board)-land-0-1-5 undefined undefined undefined undefined undefined undefined"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      />
      <div
        className="Connect(Board)-land-0-1-5 undefined undefined undefined undefined undefined undefined"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      />
      <div
        className="Connect(Board)-land-0-1-5 undefined undefined undefined undefined undefined undefined"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      />
      <div
        className="Connect(Board)-land-0-1-5 undefined undefined undefined undefined undefined undefined"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      />
      <div
        className="Connect(Board)-land-0-1-5 undefined undefined undefined undefined undefined undefined"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      />
      <div
        className="Connect(Board)-land-0-1-5 undefined undefined undefined undefined undefined undefined"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      />
      <div
        className="Connect(Board)-land-0-1-5 undefined undefined undefined undefined undefined undefined"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      />
      <div
        className="Connect(Board)-land-0-1-5 undefined undefined undefined undefined undefined undefined"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      />
      <div
        className="Connect(Board)-land-0-1-5 undefined undefined undefined undefined undefined undefined"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      />
      <div
        className="Connect(Board)-land-0-1-5 undefined undefined undefined undefined undefined undefined"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      />
      <div
        className="Connect(Board)-land-0-1-5 undefined undefined undefined undefined undefined undefined"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      />
      <div
        className="Connect(Board)-land-0-1-5 undefined undefined undefined undefined undefined undefined"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      />
      <div
        className="Connect(Board)-land-0-1-5 undefined undefined undefined undefined undefined undefined"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      />
      <div
        className="Connect(Board)-land-0-1-5 undefined undefined undefined undefined undefined undefined"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      />
      <div
        className="Connect(Board)-land-0-1-5 undefined undefined undefined undefined undefined undefined"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      />
      <div
        className="Connect(Board)-land-0-1-5 undefined undefined undefined undefined undefined undefined"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      />
      <div
        className="Connect(Board)-land-0-1-5 undefined undefined undefined undefined undefined undefined"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      />
      <div
        className="Connect(Board)-land-0-1-5 undefined undefined undefined undefined undefined undefined"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      />
      <div
        className="Connect(Board)-land-0-1-5 undefined undefined undefined undefined undefined undefined"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      />
      <div
        className="Connect(Board)-land-0-1-5 undefined undefined undefined undefined undefined undefined"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      />
      <div
        className="Connect(Board)-land-0-1-5 undefined undefined undefined undefined undefined undefined"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      />
      <div
        className="Connect(Board)-land-0-1-5 undefined undefined undefined undefined undefined undefined"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      />
      <div
        className="Connect(Board)-land-0-1-5 undefined undefined undefined undefined undefined undefined"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      />
      <div
        className="Connect(Board)-land-0-1-5 undefined undefined undefined undefined undefined undefined"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      />
      <div
        className="Connect(Board)-land-0-1-5 undefined undefined undefined undefined undefined undefined"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      />
      <div
        className="Connect(Board)-land-0-1-5 undefined undefined undefined undefined undefined undefined"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      />
      <div
        className="Connect(Board)-land-0-1-5 undefined undefined undefined undefined undefined undefined"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      />
      <div
        className="Connect(Board)-land-0-1-5 undefined undefined undefined undefined undefined undefined"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      />
      <div
        className="Connect(Board)-land-0-1-5 undefined undefined undefined undefined undefined undefined"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      />
      <div
        className="Connect(Board)-land-0-1-5 undefined undefined undefined undefined undefined undefined"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      />
      <div
        className="Connect(Board)-land-0-1-5 undefined undefined undefined undefined undefined undefined"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      />
      <div
        className="Connect(Board)-land-0-1-5 undefined undefined undefined undefined undefined undefined"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      />
      <div
        className="Connect(Board)-land-0-1-5 undefined undefined undefined undefined undefined undefined"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      />
      <div
        className="Connect(Board)-land-0-1-5 undefined undefined undefined undefined undefined undefined"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      />
      <div
        className="Connect(Board)-land-0-1-5 undefined undefined undefined undefined undefined undefined"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      />
      <div
        className="Connect(Board)-land-0-1-5 undefined undefined undefined undefined undefined undefined"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      />
      <div
        className="Connect(Board)-land-0-1-5 undefined undefined undefined undefined undefined undefined"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      />
      <div
        className="Connect(Board)-land-0-1-5 undefined undefined undefined undefined undefined undefined"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      />
      <div
        className="Connect(Board)-land-0-1-5 undefined undefined undefined undefined undefined undefined"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      />
      <div
        className="Connect(Board)-land-0-1-5 undefined undefined undefined undefined undefined undefined"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      />
      <div
        className="Connect(Board)-land-0-1-5 undefined undefined undefined undefined undefined undefined"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      />
      <div
        className="Connect(Board)-land-0-1-5 undefined undefined undefined undefined undefined undefined"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      />
      <div
        className="Connect(Board)-land-0-1-5 undefined undefined undefined undefined undefined undefined"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      />
      <div
        className="Connect(Board)-land-0-1-5 undefined undefined undefined undefined undefined undefined"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      />
      <div
        className="Connect(Board)-land-0-1-5 undefined undefined undefined undefined undefined undefined"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      />
      <div
        className="Connect(Board)-land-0-1-5 undefined undefined undefined undefined undefined undefined"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      />
      <div
        className="Connect(Board)-land-0-1-5 undefined undefined undefined undefined undefined undefined"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      />
      <div
        className="Connect(Board)-land-0-1-5 undefined undefined undefined undefined undefined undefined"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      />
      <div
        className="Connect(Board)-land-0-1-5 undefined undefined undefined undefined undefined undefined"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      />
      <div
        className="Connect(Board)-land-0-1-5 undefined undefined undefined undefined undefined undefined"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      />
      <div
        className="Connect(Board)-land-0-1-5 undefined undefined undefined undefined undefined undefined"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      />
      <div
        className="Connect(Board)-land-0-1-5 undefined undefined undefined undefined undefined undefined"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      />
      <div
        className="Connect(Board)-land-0-1-5 undefined undefined undefined undefined undefined undefined"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      />
      <div
        className="Connect(Board)-land-0-1-5 undefined undefined undefined undefined undefined undefined"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      />
    </div>
  </div>,
]
`;
